### ISSUE TITLE
snapgene format

### ISSUE THREAD

**Issue by mcmanuslab** [Lattice-Automation/seqviz#241]
**The bug**

The goal is to display a snapgene formated file in a browser. This does not appear to be working correctly. However, a genbank formatted vector of the same file works perfectly.

**Expected and observed behavior**

The expected behavior is to display a map and a sequence. However, the map is missing features and a title. The sequence displays, but it appears that non-DNA text also appears.

**Screenshot**

<img width="1445" alt="Screenshot 2023-11-19 at 5 14 13 PM" src="https://github.com/Lattice-Automation/seqviz/assets/51204856/8c02f196-2a5a-4c33-87e7-91a99f5fa133">

**The local environment:**

 - OS: MacOS
 - Browser: chrome
 - Version 119.0.6045.159/160

**Test code below**
<img width="728" alt="Screenshot 2023-11-19 at 5 38 06 PM" src="https://github.com/Lattice-Automation/seqviz/assets/51204856/38752b24-0596-4c6f-b39a-d73d87c6931b">

**Comment by isaacguerreir** [comment]
I think passing a `File` to be parsed using the `file` field of the `Seqviz` component is now deprecated. 

The recommendation is to use **[`seqparse`](https://github.com/Lattice-Automation/seqparse)**. It seems to me the library supports Snapgene file format. I would recommend trying two alternatives. One uses an `async` function to the change listener so you can make use of the `await` syntax on **seqparse**.

```{html}
<script src="https://unpkg.com/seqparse"></script>
document.getElementById('snapgeneFileInput').addEventListener('change', async function(event) {
<!-- some lines below -->
    var fileContent = e.target.result[0]
    var parsedFile = await seqparse(fileContent)
    window.seqviz
         .Viewer("seqvizViewer", {
              seq: parsedFile.seq,
              annotations: parsedFile.annotations,
              name: parsedFile.name,
              ....
         })
<!-- some more lines -->
```

 The second option would be to pass a callback function on the `then` function to instantiate your page.
 ```{html}
<!-- some lines below -->
    var fileContent = e.target.result[0]
    seqparse(fileContent).then((parsedFile) => {
       window.seqviz
         .Viewer("seqvizViewer", {
              seq: parsedFile.seq,
              annotations: parsedFile.annotations,
              name: parsedFile.name,
              ....
         })
    })
   
<!-- some more lines -->
```
Let us know if it works for you.

**Comment by mcmanuslab** [comment]
Thanks! If it's deprecated, I'll note that genebank formatted files display quite nicely-- it seems like a Snapgene format issue.

From a quick glance at the Seqparse GitHub repository, it seems that this library is primarily designed to be used in a Node.js environment or in backend Python applications, rather than directly in the browser like I am using Seqviz. 

Do you know whether Seqparse is feasible in a front-end browser implementation?

**Comment by isaacguerreir** [comment]
Is it possible to share your snapgene file? Possibly `seqparse` used on the
deprecated parameter file is not correct parsing your file.

**Comment by jjti** [comment]
> Is it possible to share your snapgene file? Possibly `seqparse` used on the deprecated parameter file is not correct parsing your file.

I agree this is likely (a bug in `seqparse`). If you share the file @mcmanuslab (either the exact one or a modified one w/ the same errant behavior) we can add it to our test cases and fix it in `seqparse`: https://github.com/Lattice-Automation/seqparse/tree/main/src/examples/snapgene

FWIW `seqviz` should still work file with file input... and that's unlikely to be removed (for backwards compatibility). The flag being deprecated is to signal that `seqviz` is really about visualization/presentation, and parsing sequence files is a bit more business logic than should have been in the library at the start

**Comment by guzmanvig** [comment]
Hello @mcmanuslab , is this still an issue? If so, could you share the file that was causing the issue? Thank you!
